### ISSUE TITLE
Use coordgen library for 2D depiction

### ISSUE THREAD

**Issue by ghutchis** [openbabel/openbabel#2015]
- [ ] I believe this to be a bug with Open Babel
- [X] This is a feature request

Schrodinger has contributed their [coordgen](https://github.com/schrodinger/coordgenlibs) library to RDKit for generating 2D depictions under the BSD license.

By all accounts, it produces high-quality depictions and it would be great to adopt a community library for depictions.

**Comment by ghutchis** [comment]
I'd probably mark this for a 3.1 release. Thoughts @baoilleach ?

**Comment by ghutchis** [comment]
I think the main plan would be to add `coordgen` as an optional compile and when available, it's used in preference to the `mcdlutil.cpp` code.

The main hook is in `ops/gen2d.cpp` right now, which simply calls `generateDiagram()`

**Comment by lorton** [comment]
@ghutchis - just in case it's not obvious, coordgen doesn't have a renderer in it (RDKit already had 2 of those, so we decided not to add to the party).  OpenBabel would probably still have to use whateverer renderer it's using, but coordgen could give 2D coordinates that would render better across small molecule ligands, materials compounds and macrocycles.

**Comment by ghutchis** [comment]
Yes, we also have several render options, including SVG, PNG, ASCII.. But we'd definitely like better 2D coordinates, which is why `coordgen` is appealing. There are several open issues and `mcdl` isn't really maintained anymore (e.g., adding new templates is difficult).

**Comment by lorton** [comment]
OK - sounds good.  So from what I can tell we could add coordgen as an optional dependency (probably turned on by default if Boost is present), and in that case we'd call into coordgen to populate coordinates instead of calling into generateDiagram?  Something like

```
#ifdef COORDGEN
generateDiagramCoordgen(...);  // New OB function to be written wrapping coordgen
#else 
generateDiagram(...);
#endif 
```

Are we happy with always using coordgen when it's built in, and building in by default if Boost is available?  If so then I could probably get a PR up for this relatively quickly.

**Comment by ghutchis** [comment]
Perfect - both the pseudo-code and building by default if Boost is available are fine. The Linux packing folks would probably want a Cmake option USE_SYSTEM_COORDGEN to use a pre-installed library and includes (e.g., USE_SYSTEM_RAPIDJSON as an example)

**Comment by lorton** [comment]
@ghutchis - would you mind if I just migrated towards the CMake way of doing this?  Using `find_package` to search for a globally installed one and only building the local library if it can't find a global library and it's asked for?  RDKit does this now (unfortunately the openbabel stuff I added was based off the old way of doing this).

**Comment by ghutchis** [comment]
No, of course you should use `find_package`.

**Comment by d-b-w** [comment]
I've dropped local installation of the templates.mae file in later releases of libcoordgen. There will also be an option not to use maeparser coordgen. Sounds like that's probably what would be desired here.
